integration/trafficdumper/dump.go

```diff
@@ -0,0 +1,134 @@
+package trafficdumper
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"time"
+
+	proxy "github.com/Telenav/osrm-backend/integration/pkg/trafficproxy"
+	"github.com/golang/glog"
+)
+
+// DumpFlowResponses dump flows to file/stdout.
+func (h Handler) DumpFlowResponses(flowResponses []*proxy.FlowResponse) {
+	if len(flowResponses) == 0 {
+		return
+	}
+
+	contentChan := make(chan string)
+	waitDoneChan := make(chan struct{})
+	if h.writeToFile {
+		go h.dumpToCSVFile("_flows", contentChan, waitDoneChan)
+	}
+
+	for _, flow := range flowResponses {
+		if glog.V(3) { // verbose debug only
+			glog.Infoln(flow)
+		}
+
+		if h.blockingOnly && !flow.Flow.IsBlocking() {
+			continue // ignore non-blocking flow
+		}
+
+		var csvString string
+		if h.humanFriendlyCSV {
+			csvString = flow.Flow.HumanFriendlyCSVString()
+		} else {
+			csvString = flow.Flow.CSVString()
+		}
+
+		if h.writeToStdout {
+			fmt.Println(csvString)
+		}
+		if h.writeToFile {
+			contentChan <- csvString
+		}
+	}
+
+	if h.writeToFile {
+		close(contentChan)
+		<-waitDoneChan
+	}
+}
+
+// DumpIncidentResponses dump incidents to file/stdout.
+func (h Handler) DumpIncidentResponses(incidentResponses []*proxy.IncidentResponse) {
+	if len(incidentResponses) == 0 {
+		return
+	}
+
+	contentChan := make(chan string)
+	waitDoneChan := make(chan struct{})
+	if h.writeToFile {
+		go h.dumpToCSVFile("_incidents", contentChan, waitDoneChan)
+	}
+
+	for _, incident := range incidentResponses {
+		if glog.V(3) { // verbose debug only
+			glog.Infoln(incident)
+		}
+
+		if h.blockingOnly && !incident.Incident.IsBlocking {
+			continue // ignore non-blocking incident
+		}
+
+		var csvString string
+		if h.humanFriendlyCSV {
+			csvString = incident.Incident.HumanFriendlyCSVString()
+		} else {
+			csvString = incident.Incident.CSVString()
+		}
+
+		if h.writeToStdout {
+			fmt.Println(csvString)
+		}
+		if h.writeToFile {
+			contentChan <- csvString
+		}
+	}
+
+	if h.writeToFile {
+		close(contentChan)
+		<-waitDoneChan
+	}
+}
+
+func (h Handler) dumpToCSVFile(fileTag string, sink <-chan string, done chan<- struct{}) {
+	defer close(done)
+	filePath := h.dumpFileNamePrefix + fileTag + ".csv"
+	startTime := time.Now()
+	var dumpedLines int64
+	defer func() {
+		glog.Infof("dumpToCSVFile %s takes %f seconds, total lines %d.\n",
+			filePath, time.Now().Sub(startTime).Seconds(), dumpedLines)
+	}()
+
+	// open file
+	//outfile, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0755)
```

Comment:
It's not test code only. 
Actually I'm thinking about whether we need a command line flag to control `append` or `overwrite` when dump to file. I choose to use `append` as default , because it will be useful when we hope to dump traffic data into a big file in `delta` mode. And I did NOT support the flag at the end. So when use the `trafficproxy-cli` to dump traffic data, it's necessary to remove old files if use same file name. 
I leave the code here, maybe we'll need it further.

Reply:
Thanks for the explaining.  Got your point.
May be in the future we could leave some comments here to let user under stand there could be a different option.